src/calc.y

```diff
@@ -17,20 +21,29 @@ Term -> Result<AstNode, ()>:
 Factor -> Result<AstNode, ()>:
     "(" Expr ")" { $2 }
     | "INTEGER" { 
-        match $1.map_err(|err| format!("Parsing Error: {}", err)) {
-            Ok(s) => {
-              let s = $lexer.span_str(s.span());
-              match s.parse::<u64>() {
-                  Ok(n_val) => Ok(AstNode::Number{ value: n_val }),
-                  Err(_) => Err(())
-              }
-            }
-            Err(_) => Err(())
-        }
+        let v = $1.map_err(|_| ())?;
+        parse_int($lexer.span_str(v.span()))
       }
+    | "ID" { 
+       let v = $1.map_err(|_| ())?;
+       Ok(AstNode::ID{ value: $lexer.span_str(v.span()).to_string() })
+    }
     ;
 
+
 Unmatched -> ():
       "UNMATCHED" { };
 %%
+
 use crate::ast::AstNode;
+
+
+fn parse_int(s: &str) -> Result<AstNode, ()> {
+    match s.parse::<u64>() {
+        Ok(n_val) => Ok(AstNode::Number{ value: n_val }),
+        Err(_) => {
+            eprintln!("{} cannot be represented as a u64", s);
```

Comment:
That should eventually be an error rippled up.

Reply:
will be added as separate PR, based on https://github.com/Pavel-Durov/pavel.yaiwr/tree/error-handling